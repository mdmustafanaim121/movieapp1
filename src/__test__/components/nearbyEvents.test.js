import React from "react";
import renderer from "react-test-renderer";
import NearbyEvents from "../../component/nearbyEvents";

describe("Testing Nearby Events component", () => {
  test("initial Snapshot", () => {
    const movies = [
      {
        name: "Black Panther",
        language: "ENGLISH",
        rate: 4.5,
        type: "Action Adventure Fantasy",
        imageUrl: "https://image.ibb.co/f0hhZc/bp.jpg",
      },
      {
        name: "Black Cat",
        language: "ENGLISH",
        rate: 4.5,
        type: "Action Adventure Fantasy",
        imageUrl: "https://image.ibb.co/f0hhZc/bp.jpg",
      },
    ];
    const isLoading = false;
    const error = "";
    const history = window.history;

    const test1 = renderer
      .create(
        <NearbyEvents
          movies={movies}
          loading={isLoading}
          history={history}
          error={error}
        />
      )
      .toJSON();
    expect(test1).toMatchSnapshot();
  });

  test("expect error message", () => {
    const isLoading = false;
    const error = {
      message: "404 url not found",
    };
    const history = window.history;
    const movies = null;
    const result = renderer
      .create(
        <NearbyEvents
          loading={isLoading}
          error={error}
          history={history}
          movies={movies}
        />
      )
      .toJSON();
    expect(result.children[3]).toBe("404 url not found");
  });

  test("expect loading", () => {
    const isLoading = true;
    const error = "";
    const history = window.history;
    const movies = null;
    const result = renderer
      .create(
        <NearbyEvents
          loading={isLoading}
          error={error}
          history={history}
          movies={movies}
        />
      )
      .toJSON();
    expect(result.props.className).toBe("loading");
  });
});
