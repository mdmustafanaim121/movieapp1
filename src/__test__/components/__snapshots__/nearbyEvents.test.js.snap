// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Nearby Events component initial Snapshot 1`] = `
<div>
  <canvas
    className="particles-bg-canvas-self"
    onMouseDown={[Function]}
    style={
      Object {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": -1,
      }
    }
  />
  <h1
    className="heading1"
  >
    Nearby Events
  </h1>
  <div
    className="d1"
  >
    <div
      className="sc-jSFjdj jvXOvp rec rec-carousel-wrapper "
      style={Object {}}
    >
      <div
        className="sc-eCApnc jCKguF rec rec-carousel"
        size={
          Object {
            "height": 0,
          }
        }
        style={
          Object {
            "height": 0,
          }
        }
      >
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-left"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❮
        </button>
        <div
          className="sc-dlnjwi dTGAUq rec rec-slider-container"
        >
          <div
            className="sc-hKFxyN gXEQKd rec rec-slider"
            style={
              Object {
                "left": "0px",
                "right": "auto",
                "top": "auto",
                "transition": "all 500ms ease",
              }
            }
          >
            <div
              className="rec rec-swipable"
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            />
          </div>
        </div>
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-right"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❯
        </button>
      </div>
      <div
        className="sc-iCoGMd xzcEF rec rec-pagination"
      >
        <button
          className="sc-gKAaRy ejXxNI rec rec-dot rec rec-dot_active"
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        />
      </div>
    </div>
  </div>
</div>
`;
